addons/stock/models/stock_move_line.py

```diff
@@ -508,14 +508,15 @@ def _free_reservation(self, product_id, location_id, quantity, lot_id=None, pack
                 ('product_qty', '>', 0.0),
                 ('id', 'not in', ml_to_ignore.ids),
             ]
-            oudated_candidates = self.env['stock.move.line'].search(oudated_move_lines_domain)
+            current_picking_first = lambda cand: cand.picking_id != self.move_id.picking_id
```

Comment:
lambda's always give me headaches... What is happening here?

Reply:
It's just a backport of an [approved Odoo 12.0 PR](https://github.com/odoo/odoo/pull/35021).

The code is indeed a bit cryptic: in this line they're defining a sort function that, in the next line, is applied after searching for move lines. This sort function then outputs `True` if the picking doesn't match, and `False` if the picking matches, and since `False` comes before `True`, so in other words, after the move lines are searched, it will sort such that all the move lines belonging to the current picking come first, and the rest comes after. Thereby solving the bug that was reported, in which move lines were scrapped from other pickings than the current.